app/controllers/incoming_webhooks/mailchimp_unsubscribes_controller.rb

```diff
@@ -0,0 +1,30 @@
+class IncomingWebhooks::MailchimpUnsubscribesController < ApplicationController
+  class InvalidListID < StandardError; end
+
+  LIST_MAPPINGS = {
+    mailchimp_newsletter_id: :email_newsletter,
+    mailchimp_sustaining_members_id: :email_membership_newsletter,
+    mailchimp_tag_moderators_id: :email_tag_mod_newsletter,
+    mailchimp_community_moderators_id: :email_community_mod_newsletter
```

Comment:
from what i can tell, we're not using most of these keys; what do we need them for? (just trying to get context on this)

Reply:
This is the mapping of option names in the `SiteConfig` model to newsletter types. We're using it here:

```ruby
    list_id = params.dig(:data, :list_id)
    key = LIST_MAPPINGS.keys.detect { |k| SiteConfig.public_send(k) == list_id }
```

Once we have mapped from list id to attribute name, we use in the update `user.update(email_type => false)`.

Does that clear it up?